### PR TITLE
[exporter/clickhouse] use observed timestamp in logs

### PR DESCRIPTION
**Pull request by pimeys**
**Description:** Some senders, such as the rust-opentelemetry do _not_ send timestamp in their logs. In these cases, if the timestamp is not set, we must use the observed timestamp instead.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34150

**Testing:** Added a test which does not define timestamp

### CODE REVIEW

**Review by hanjm — APPROVED** [review]
code LGTM, Thanks. please add changelog.

**Review by SpencerTorres — APPROVED** [review]
Nice find, appreciate the contribution. Add a changelog entry and this should be good to go.

Thanks!

**Comment by pimeys** [comment]
I added the changelog and amended. Should be good now.

**Comment by SpencerTorres** [comment]
cc @dmitryax @crobert-1

File: .chloggen/use-observed-timestamp-if-timestamp-is-missing.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+note: Use observed timestamp if timestamp is zero
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [34150]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+  Some opentelemetry libraries do not send timestamp for logs, but they should always send
```
Comment (crobert-1) [inline .chloggen/use-observed-timestamp-if-timestamp-is-missing.yaml:19]:
```suggestion
  Some OpenTelemetry libraries do not send timestamp for logs, but they should always send
```

File: .chloggen/use-observed-timestamp-if-timestamp-is-missing.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  | the observed timestamp. In these cases the clickhouse exporter just stored a zero timestamp
```
Comment (crobert-1) [inline .chloggen/use-observed-timestamp-if-timestamp-is-missing.yaml:20]:
```suggestion
  | the observed timestamp. In these cases the ClickHouse exporter just stored a zero timestamp
```

**Comment by SpencerTorres** [comment]
@pimeys looks like there's a couple nits to the changelog. Let me know if you can resolve those so we can get this merged 👍

**Review by crobert-1 — APPROVED** [review]

**Review by Frapschen — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**